SignalServiceKit/src/Messages/Interactions/TSQuotedMessage.h

```diff
@@ -0,0 +1,39 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import <SignalServiceKit/TSYapDatabaseObject.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface TSQuotedMessage : TSYapDatabaseObject
+
+@property (nonatomic, readonly) uint64_t timestamp;
+@property (nonatomic, readonly) NSString *authorId;
+
+// This property should be set IFF we are quoting a text message.
+@property (nullable, nonatomic, readonly) NSString *body;
+
+// This property should be set IFF we are quoting a attachment message.
+@property (nullable, nonatomic, readonly) NSString *sourceFilename;
+// This property can be set IFF we are quoting a attachment message, but it is optional.
+@property (nullable, nonatomic, readonly) NSData *thumbnailData;
+// This is a MIME type.
+//
+// This property should be set IFF we are quoting a attachment message.
+@property (nullable, nonatomic, readonly) NSString *contentType;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+- (instancetype)initWithTimestamp:(uint64_t)timestamp
```

Comment:
I wasn't going to bother applying the same pattern here since `TSQuotedMessage` (like most of our classes) doesn't have a deep inheritance hierarchy, so this issue doesn't apply.